modules/affile/file-reader.c

```diff
@@ -77,6 +77,13 @@ _recover_state(LogPipe *s, GlobalConfig *cfg, LogProtoServer *proto)
     }
 }
 
+static gboolean
+_can_check_eof(gint fd)
+{
+  struct stat st;
+  return fstat(fd, &st) == 0 && !S_ISFIFO(st.st_mode);
```

Comment:
i know the code is full of expressions like this, but i'd prefer to use anything but this form, as this is extremely unreadable, especially in cases when the first letter of the following expression makes it even harder to read

anything is better than the `!` operator in these cases, e.g.

FALSE == expr
or 
expr == 0
etc.

Reply:
Ichanged it to  `fstat(fd, &st) == 0 && S_ISFIFO(st.st_mode) != 0`